tests/system/robot/chromeTests.py

```diff
@@ -1705,6 +1705,147 @@ def test_tableNavigationWithMergedColumns():
 	_asserts.strings_match(actualSpeech, "row 1  column 2  b 1")
 
 
+def test_tableSayAll():
+	""" Tests that table sayAll commands and speak row/column commands work.
+	Also that they work correctly with merged cells.
+	Refer to #13469.
+	"""
+	_chrome.prepareChrome("""
+		<p>Hello, world!</p>
+		<table border=3>
+			<tr>
+				<td>A1</td>
+				<td>B1</td>
+				<td rowspan=2>C1+C2</td>
+				<td>D1</td>
+				<td>E1</td>
+			</tr>
+			<tr>
+				<td>A2</td>
+				<td>B2</td>
+				<td>D2</td>
+				<td>E2</td>
+			</tr>
+			<tr>
+				<td colspan=2>A3+B3</td>
+				<td>C3</td>
+				<td colspan=2>D3+E3</td>
+			</tr>
+			<tr>
+				<td>A4</td>
+				<td>B4</td>
+				<td colspan=2 rowspan=2>C4+D4+<br>C5+D5</td>
+				<td>E4</td>
+			</tr>
+			<tr>
+				<td>A5</td>
+				<td>B5</td>
+				<td>E5</td>
+			</tr>
+		</table>
+		<p>Bye-bye, world!</p>
+
+	""")
+
+	# Jump to table
+	actualSpeech = _chrome.getSpeechAfterKey("t")
+	_asserts.strings_match(actualSpeech, "table  with 5 rows and 5 columns  row 1  column 1  A 1")
+
+	def jumpToB2():
+		_chrome.getSpeechAfterKey("Control+Alt+PageUp")
+		_chrome.getSpeechAfterKey("Control+Alt+Home")
+		_chrome.getSpeechAfterKey("Control+Alt+RightArrow")
+		actualSpeech = _chrome.getSpeechAfterKey("Control+Alt+DownArrow")
+		_asserts.strings_match(actualSpeech, "row 2  B 2")
+
+	def jumpToA3():
+		_chrome.getSpeechAfterKey("Control+Alt+PageUp")
+		_chrome.getSpeechAfterKey("Control+Alt+Home")
+		_chrome.getSpeechAfterKey("Control+Alt+DownArrow")
+		actualSpeech = _chrome.getSpeechAfterKey("Control+Alt+DownArrow")
+		_asserts.strings_match(actualSpeech, "row 3  column 1  through 2  A 3 plus B 3")
+
+	def jumpToD5():
+		_chrome.getSpeechAfterKey("Control+Alt+PageUp")
+		_chrome.getSpeechAfterKey("Control+Alt+End")
+		_chrome.getSpeechAfterKey("Control+Alt+LeftArrow")
+		_chrome.getSpeechAfterKey("Control+Alt+DownArrow")
+		_chrome.getSpeechAfterKey("Control+Alt+DownArrow")
+		actualSpeech = _chrome.getSpeechAfterKey("Control+Alt+DownArrow")
+		_asserts.strings_match(
+			actualSpeech,
+			"row 4  column 3  through row 5 column 4  C 4 plus D 4 plus  C 5 plus D 5"
+		)
+
+	if True:

```

Comment:
I suggest splitting these `if` blocks up into separate tests eg `_test_SayAllReadEntireColumn`.

`test_tableSayAll` could call all of these.